### PR TITLE
Confusing error when request is targetted for an invalid tenant id #1792

### LINKED ISSUE #1792: Confusing error when request is targetted for an invalid tenant id
**Describe the bug**
Requests for an invalid tenant are coming back with a response like this:
```
{"resourceType":"OperationOutcome","id":"c0-a8-1-da-7c8d79ec-4bc3-4d6b-940b-d755e1bb4f24","issue":[{"severity":"fatal","code":"exception","details":{"text":"FHIRPersistenceException: fhirServer/persistence/factoryClassname is configured incorrectly"}}]}
```

**To Reproduce**
1. deploy the server with the default connfig
2. make a request with a header like `X-FHIR-TENANT-ID: dummy`

**Expected behavior**
The request should come back with a clear error message that states that the tenant wasn't valid

**Additional context**
In #639 we decided that all tenants should have their own datasources (and should never "fall back" to the default one).
Based on this, we should be able to check for the presence of a tenant up front (based on the existence of the corresponding tenant config directory).  Maybe we should put it right in the FHIRRestServletFilter?

### PR DESCRIPTION
**Pull request by prb112**
- Disambiguates the Error Messages that are bubbled up through the
Persistence Layer
- Add Integration Test

Closes #1792 

Signed-off-by: Paul Bastide <pbastide@us.ibm.com>

### CODE REVIEW

File: fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/cache/FHIRPersistenceJDBCTenantCache.java [hunk 1]
```diff
@@ -74,14 +74,16 @@ protected FHIRPersistenceJDBCCache createCache(String cacheKey) {
                 int externalValueCacheSize = pg.getIntProperty("externalValueCacheSize", 100000);
                 return FHIRPersistenceJDBCCacheUtil.create(externalSystemCacheSize, externalValueCacheSize);
             }
+        } catch(IllegalStateException ise) {
```
Comment (punktilious) [inline fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/cache/FHIRPersistenceJDBCTenantCache.java:77]: space after catch

File: fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/cache/FHIRPersistenceJDBCTenantCache.java [hunk 2]
```diff
@@ -33,14 +33,14 @@
...
     public FHIRPersistenceJDBCCache getCacheForTenantAndDatasource() {
         final String cacheKey = getCacheKeyForTenantDatasource();
-        
+
         // Obtain the cache for the current tenant/datasource, creating a new one if need be
         FHIRPersistenceJDBCCache result = cacheMap.computeIfAbsent(cacheKey, (k) -> createCache(k));
         if (result == null) {
             // exception handled inside createCache (which will have been logged)
-            throw new IllegalStateException("createCache failed");
+            throw new IllegalStateException("createCache failed " + cacheKey);
```
Comment (punktilious) [inline fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/cache/FHIRPersistenceJDBCTenantCache.java:41]: Worth noting it is intentional that this message is included in the request response?

**Review by punktilious — APPROVED** [review]
LGTM with a couple of minor comments

File: fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/cache/FHIRPersistenceJDBCTenantCache.java (hunk 2 above)
Comment (prb112) [reply fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/cache/FHIRPersistenceJDBCTenantCache.java:41]: Yes, that's right I'll add a comment

Comment (prb112) [reply fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/cache/FHIRPersistenceJDBCTenantCache.java]: comment is added

**Review by lmsurpre — APPROVED** [review]
LGTM

**Pull request merged**